src/gui/mzroll/background_peaks_update.cpp

```diff
@@ -446,7 +448,8 @@ void BackgroundPeakUpdate::updateGroups(QList<PeakGroup> &groups,
         {
             for(Peak& peak :  group.peaks)
             {
-                if (eic->getSample() == peak.getSample()){
+                if (eic->getSample()->getSampleName() ==
+                    peak.getSample()->getSampleName()){
```

Comment:
Why resort to using the sample's name and not its pointer? The effect should be the same no? In fact, maybe the earlier way of checking was more robust.

Reply:
yeaa.. sure I will make this change.